oeps/architectural-decisions/oep-0065/decisions/0001-unified-platform-repository.rst

```diff
@@ -0,0 +1,109 @@
+OEP-65 Create a unified platform repository
+###########################################
+
+Status
+******
+
+Accepted
+
+Summary
+*******
+
+This ADR proposes combining a number of repositories into a single, unified frontend platform library as a migration strategy for implementing OEP-65.
+
+Context
+*******
+
+OEP-65 proposes adopting webpack module federation for Open edX micro-frontends (MFEs) as a means to enable *runtime module loading* and *shared dependencies* in the Open edX frontend.  The OEP lays out a series of changes necessary to enable these capabilities, which it refers to as building a "reference implementation" of runtime module loading and shared dependencies.  This reference implementation is effectively a new underlying architecture for our frontend. This ADR refers to this new architecture as the "module MFE" architecture, as opposed to the historical "application MFE" architecture that has existed prior to OEP-65.
+
+The bulk of the work to build OEP-65's reference implementation is related to the *library repositories* that MFEs depend on:
+
+* frontend-build
+* frontend-platform
+* frontend-plugin-framework
+* frontend-component-header
+* frontend-component-footer
+
+It will also involve the creation of a new *shell application* to initialize the frontend at runtime and load existing MFEs as modules.
+
+There are several related concerns which influence our strategy for migrating these library repositories to the new module MFE architecture.
+
+Significant API changes
+=======================
+
+The creation of the shell application and the move toward using plugins and module federation is expected to significantly change our libraries' API surface as the shell takes responsibility for much of what ``frontend-platform`` provides to MFEs in the application MFE architecture.
+
+The development environment will also change, as we want to be able to run a single development server that loads the MFE under development into the shell, which means alternatives to frontend-build's existing helpers will need to be created.
+
+The header and footer will be owned by the shell and need not be libraries any more (caveat: maintaining good ways to customize them), and we expect ``frontend-plugin-framework`` to be folded into ``frontend-platform`` to unify the two, as they will be intimately tied to each other going forward.
```

Comment:
Would it make sense to split this up and have "Significant API Changes" subsections in other relevant sections? I think these might benefit from context.

Reply:
Hm, I'm not sure how I would change this/am not quite seeing the lack of context...  The previous section talks about the libraries involved, and this section describes the significant API changes to them.  The other sections below aren't talking about those libraries specifically, but other relevant context.

Reply:
Coming back to this now I think I have a better sense of where I was coming from, and a better possible suggestion.

I think I was having a hard time parsing *where* the APIs being changed were, and I think it would be clearer to me if the API changes were split up by repo so it looks something like 

```suggestion
Significant API changes
=======================

* ``frontend-platform``


  * The creation of the shell application and the move toward using plugins and module federation is expected to significantly change our libraries' API surface as the shell takes responsibility for much of what ``frontend-platform`` provides to MFEs in the application MFE architecture.


* ``frontend-app-*``


  * The development environment will change, as we want to be able to run a single development server that loads the MFE under development into the shell.


* ``frontend-build``


  * To support the aforementioned single development server, alternatives to frontend-build's existing helpers will need to be created.


* ``frontend-component-header``/``frontend-component-footer``


  * The header and footer will be owned by the shell and need not be libraries any more (caveat: maintaining good ways to customize them).


* ``frontend-plugin-framework``


  * We expect ``frontend-plugin-framework`` to be folded into ``frontend-platform`` to unify the two, as they will be intimately tied to each other going forward.
```

Reply:
Nice!  I like it.  Thanks for explaining. :)